sdk/core/Azure.Core/tests/compatibility/ExpectedAotWarnings.txt

```diff
@@ -12,15 +14,26 @@ Azure\.Core.src.Shared.DiagnosticScope\.cs\(\d*\): AOT analysis warning IL3050:
 Azure\.Core.src.DynamicData.DynamicDataProperty\.cs\(\d*\): Trim analysis warning IL2026: Azure\.Core\.Serialization\.DynamicDataProperty\.MetaObject\.BindGetMember\(GetMemberBinder\): Using member 'System\.Linq\.Expressions\.Expression\.Property\(Expression,String\)' which has 'RequiresUnreferencedCodeAttribute'
 ILC : Trim analysis warning IL2026: System\.BinaryData\.BinaryData\(Object,JsonSerializerOptions,Type\): Using member 'System\.Text\.Json\.JsonSerializer\.SerializeToUtf8Bytes\(Object,Type,JsonSerializerOptions\)' which has 'RequiresUnreferencedCodeAttribute'
 ILC : AOT analysis warning IL3050: System\.BinaryData\.BinaryData\(Object,JsonSerializerOptions,Type\): Using member 'System\.Text\.Json\.JsonSerializer\.SerializeToUtf8Bytes\(Object,Type,JsonSerializerOptions\)' which has 'RequiresDynamicCodeAttribute'
 ILC : Trim analysis warning IL2026: System\.BinaryData\.ToObjectFromJson<T>\(JsonSerializerOptions\): Using member 'System\.Text\.Json\.JsonSerializer\.Deserialize\(ReadOnlySpan`1<Byte>,Type,JsonSerializerOptions\)' which has 'RequiresUnreferencedCodeAttribute'
 ILC : AOT analysis warning IL3050: System\.BinaryData\.ToObjectFromJson<T>\(JsonSerializerOptions\): Using member 'System\.Text\.Json\.JsonSerializer\.Deserialize\(ReadOnlySpan`1<Byte>,Type,JsonSerializerOptions\)' which has 'RequiresDynamicCodeAttribute'
-System\.Linq\.Expressions.src.System.Linq.Expressions.MethodCallExpression\.cs\(\d*\): AOT analysis warning IL3050: System\.Linq\.Expressions\.Expression\.ApplyTypeArgs\(MethodInfo,Type\[\]\): Using member 'System\.Reflection\.MethodInfo\.MakeGenericMethod\(Type\[\]\)'
-System\.Linq\.Expressions.src.System.Dynamic.Utils.TypeUtils\.cs\(\d*\): AOT analysis warning IL3050: System\.Dynamic\.Utils\.TypeUtils\.GetNullableType\(Type\): Using member 'System\.Type\.MakeGenericType\(Type\[\]\)' which has 'RequiresDynamicCodeAttribute'
-System\.Linq\.Expressions.src.System.Linq.Expressions.BinaryExpression\.cs\(\d*\): AOT analysis warning IL3050: System.Linq.Expressions.Expression.GetResultTypeOfShift\(Type,Type\): Using member 'System.Type.MakeGenericType\(Type\[\]\)' which has 'RequiresDynamicCodeAttribute'
-System\.Net\.Http.src.System.Net.Http.DiagnosticsHandler\.cs\(\d*\): AOT analysis warning IL3050: System\.Net\.Http\.DiagnosticsHandler\.Write<T>\(DiagnosticSource,String,!!0\): Using member 'System\.Diagnostics\.DiagnosticSource\.Write\(String,Object\)' which has 'RequiresDynamicCodeAttribute'
\ No newline at end of file
+Azure\.Core.src.DynamicData.DynamicData\.DynamicMetaObject\.cs\(\d*\): Trim analysis warning IL2026: Azure\.Core\.Serialization\.DynamicData\.MetaObject\.BindGetIndex\(GetIndexBinder,DynamicMetaObject\[\]\): Using member 'Azure\.Core\.Serialization\.DynamicData\.GetViaIndexerMethod' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code
```

Comment:
Adding these just to get the pipeline green. Will follow up - https://github.com/Azure/azure-sdk-for-net/issues/44296